pkgs/applications/window-managers/hyprwm/hyprshade/default.nix

```
@@ -0,0 +1,32 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pdm
+, pdm-backend
+, more-itertools
+, click
+}:
+
+buildPythonPackage rec {
+  pname = "hyprshade";
+  version = "0.9.3";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "loqusion";
+    repo = "hyprshade";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ou072V9nZUqf5DEolkMQy979SjaZs4iOuoszw50k4Y8=";
+  };
+
+  propagatedBuildInputs = [ pdm pdm-backend more-itertools click ];
```

```suggestion
  nativeBuildInputs = [
    pdm-backend
  ];

  propagatedBuildInputs = [ more-itertools click ];
```
`pdm-backend` is a build time only dependency.